src/libmongoc/tests/test-mongoc-handshake.c

```diff
@@ -220,18 +220,20 @@ _handshake_check_os (bson_t *doc)
    ASSERT (val);
    ASSERT (strlen (val) > 0);
 
-   /* Check os version valid */
-   ASSERT (bson_iter_find (&inner_iter, "version"));
-   ASSERT (BSON_ITER_HOLDS_UTF8 (&inner_iter));
-   val = bson_iter_utf8 (&inner_iter, NULL);
-   _check_os_version_valid (val);
+   /* Check os version is valid if present (may be truncated) */
+   if (bson_iter_find (&inner_iter, "version")) {
+      ASSERT (BSON_ITER_HOLDS_UTF8 (&inner_iter));
+      val = bson_iter_utf8 (&inner_iter, NULL);
+      _check_os_version_valid (val);
+   }
```

Comment:
Since `_override_host_platform_os` is used to avoid truncation, can `version` or `architecture` be assumed to exist? If so, suggest reverting the added `if` checks to improve test coverage.